src/main/java/org/assertj/core/api/ThrowableCollector.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import java.util.List;
+
+public interface ThrowableCollector {
+
+	void collectError(Throwable t);
```

Comment:
the existing `ErrorCollector` has an `addError(Throwable error)` but actually it think it could have been `addError(AssertionError error)`. Let's try here to define void `collectAssertionError(AssertionError t);`.

rename `ThrowableCollector` to `AssertionErrorCollector` to be consistent.

Reply:
I'm not sold on this one.

The point of soft assertions is that if one of them fails it doesn't stop the others from running. This is useful regardless of whether the test failed due to an error (ie, an exception other than `AssertionError`) or an assertion failure (ie, an `AssertionError`).

I have personally had experience when using soft assertions that I had an error and it masked a whole bunch of soft assertion failures. I was lulled into a false sense of security thinking that I had only the one error - when I fixed it, a whole bunch of assertion failures appeared. :smile:

On the other hand, it will be easier to restrict this to `AssertionError` now and find we need to expand it to include `Throwable`, rather than include `Throwable` now and find we need to restrict it to `AssertionError`.

In the end I guess this is your call - I'll go with what you want, I just wanted to put forward an alternative viewpoint for leaving it how it is.

Reply:
Fair point, to do that `ErrorCollector` would have to change, at the moment its `intercept` method only catch `AssertionError`.

I would rather go with `AssertionError` for now to keep changes in this PR not to be too big.

I have created https://github.com/joel-costigliola/assertj-core/issues/1821 to think about it further.

Reply:
Actually @joel-costigliola, I started going down this path and it's bit more complicated.

`List<Throwable> errorsCollected()` was an existing public method of `AbstractSoftAssertions` - I simply copied that signature into `Throwable/AssertionErrorCollector`. If I change the signature of `AssertionErrorCollector`, I'll probably still have to leave the old method in place for backward compatibility.

Reply:
This has been done. Depending on the outcome of #1821, we may want to deprecate `errorsCollected()`.